src/modules/Account/account.service.ts

```diff
@@ -69,6 +80,63 @@ export class AccountService {
     // todo: logger ({ location: 'SRC:MODULES:ACCOUNT:ACCOUNT_SERVICE::CREATE_ACCOUNT' });
   }
 
+  async resetPassword(
+    resetPasswordInput: ResetPasswordInput
+  ): Promise<ResetPasswordOutput> {
+    const hashedEmail = await hashDataAsync(
+      resetPasswordInput.email,
+      process.env.SALT_DATA
+    );
+
+    const accountExists = await this.accountRepository.alreadyExists(
+      hashedEmail
+    );
+    if (!accountExists) throw new AccountNotFoundError();
+    const account = await this.accountRepository.findAccountByEmail(
+      hashedEmail
+    );
+
+    // todo: check if user code has expired
```

Comment:
Pode adicionar o tempo de espiração do token no esquema `ResetPasswordToken`, assim, quando você for buscar um token, pode pegar um token onde o tempo de expiração ainda não passou `expiresDateTime > currentDateTime `. [Prisma gt](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#gt)

Reply:
Outro ponto, como o id da conta é único no esquema, podes usar `upsert` do prisma para que ele crie ou atualize o token de recuperar conta caso o token já exista já exista.

Isso também vale caso o usuário peça outro código e esse ainda exista na base de dados.

Reply:
Não implementei o tempo de expiração porque não ia dar tempo e não era requisito do backlog.

Reply:
será que realmente é necessário que esteja, não é algo que já faz parte do sistema? assim como a expiração de um token de sessão?